### PR TITLE
Function Documentation ReDo

### PR DESCRIPTION
**Pull request by elimillera**
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.  

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs.  We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

Draft for now while I work through all of the updates

Good number of updates to functions:
- Updated the params to inherit from a single spot where possible
- Added Metadata and messaging sections for each function.
- Added options documentation on package help page

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description.  Can be removed or left blank if changes are minor/self-explanatory.
- [x] Check that your Pull Request is targeting the `devel` branch, Pull Requests to `main` should use the [Release Pull Request Template](https://github.com/atorus-research/xportr/tree/94_pr_template/.github/PULL_REQUEST_TEMPLATE)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/).  Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by elimillera** [comment]
> Is this branched off of the `main` branch rather than `devel`?

It wasn't but it looks like a git merge went wrong and merged in what we have from main

File: R/length.R [hunk 1]
```diff
@@ -1,18 +1,49 @@
...
-#' @param verbose The action the function takes when a length isn't found in
-#'   metadata. Options are 'stop', 'warn', 'message', and 'none'
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
+#' @param metadata A data frame containing dataset/variable level metadata. See
+#'   'Metadata' section for details.
+#' @param domain Appropriate CDSIC dataset name, e.g. ADAE, DM. Used to subset
+#'   the metadata object. If none is passed, then name of the dataset passed as
+#'   .df will be used.
+#' @param verbose The action this function takes when an action is taken on the
+#'   dataset or function validation finds an issue. See 'Messaging' section for
```
Comment (bms63) [inline R/length.R:15]: Is there a way to internally link the Messaging and Metadata sections? @vedhav or @averissimo any thoughts?

File: R/length.R [hunk 2]
```diff
@@ -1,18 +1,49 @@
...
+#'
+#' @section Messaging: `length_log` is the primary messaging tool for
+#'   `xportr_length`. If there are any columns present in the '.df' that are not
+#'   noted in the metadata, they cannot be assigned a length and a message will
+#'   be generated noting the number or variables that have not been assigned a
+#'   length.
+#'
+#'   If variables were not found in the metadata and the value passed to the
+#'   'verbose' argument is 'stop', 'warn', or 'message', a message will be
+#'   generated detailing the variables that were missing in metadata.
```
Comment (bms63) [inline R/length.R:28]:
```suggestion
#'   generated detailing the variables that were missing in the metadata.
```

File: R/length.R [hunk 3]
```diff
@@ -1,18 +1,49 @@
...
+#'
+#' @section Metadata: The argument passed in the 'metadata' argument can either
+#'   be a metacore object, or a data.frame containing the data listed below. If
```
Comment (bms63) [inline R/length.R:31]:
```suggestion
#'   be a `{metacore}` object, or a data.frame containing the data listed below. If
```
Then it links to the metacore pkgdown site

File: R/length.R [hunk 4]
```diff
@@ -1,18 +1,49 @@
...
+#'   metacore is used, no changes to options are required.
+#'
+#'   For data.frame 'metadata' arguments three columns must be present:
+#'
+#'   1) Domain Name - passed as the 'xportr.domain_name' option. Default:
+#'   "dataset". This is the column subset by the 'domain' argument in the
+#'   function. 2) Variable Name - passed as the 'xportr.variable_name' option.
+#'   Default: "variable". This is used to match columns in '.df' argument and
+#'   the metadata. 3) Variable Label - passed as the 'xportr.length' option.
+#'   Default: "length". These numeric values to update the 'width' attribute of
```
Comment (bms63) [inline R/length.R:41]: These itemized list is a bit wonky 
image

File: R/length.R [hunk 5]
```diff
@@ -1,18 +1,49 @@
...
+#'   the column. This is passed to `haven::write` to note the variable length.
+#'
 #'
 #' @return Data frame with `SASlength` attributes for each variable.
-#' @family metadata functions
-#' @seealso [xportr_label()], [xportr_df_label()] and [xportr_format()]
+#'
 #' @export
 #'
 #' @examples
```
Comment (bms63) [inline R/length.R:49]: Could we also create simple examples that use `verbose = "warn"` in each of our core functions?

File: R/type.R [hunk 6]
```diff
@@ -1,18 +1,51 @@
...
-#'   dataset passed in `.df`.
-#' @param verbose The action the function takes when a variable isn't typed
-#'   properly. Options are 'stop', 'warn', 'message', and 'none'
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
+#' @inheritParams xportr_length
+#'
+#' @section Messaging: `type_log` is the primary messaging tool for
+#'   `xportr_type`. The number of column types that mismatch the reported type
+#'   in the metadata, if any, is reported by by `xportr_type`. If there are any
+#'   type mismatches, and the 'verbose' argument is 'stop', 'warn', or
```
Comment (bms63) [inline R/type.R:15]:
```suggestion
#' @section Messaging: `type_log()` is the primary messaging tool for
#'   `xportr_type()`. The number of column types that mismatch the reported type
#'   in the metadata, if any, is reported by by `xportr_type()`. If there are any
#'   type mismatches, and the 'verbose' argument is 'stop', 'warn', or
```

File: R/type.R [hunk 7]
```diff
@@ -1,18 +1,51 @@
...
+#'   'message', the type mismatches will be detailed the actual type and the
+#'   type reported in the metadata.
```
Comment (bms63) [inline R/type.R:17]: This phrase needs a little polish

File: R/type.R [hunk 8]
```diff
@@ -1,18 +1,51 @@
...
+#'
+#' @section Metadata: The argument passed in the 'metadata' argument can either
+#'   be a metacore object, or a data.frame containing the data listed below. If
+#'   metacore is used, no changes to options are required.
+#'
+#'   For data.frame 'metadata' arguments three columns must be present:
```
Comment (bms63) [inline R/type.R:23]: should this be four columns?

File: R/type.R [hunk 9]
```diff
@@ -1,18 +1,51 @@
...
+#'   4) Variable Type - passed as the 'xportr.type_name'. Default: "type". This
+#'   is used to note the XPT variable "type" options are numeric or character.
+#'
+#'   5) (Option only) Character Types - The list of classes that should be
+#'   explicitly coerced to a XPT Character type. Default: c( "character",
+#'   "char", "text", "date", "posixct", "posixt", "datetime", "time",
+#'   "partialdate", "partialtime", "partialdatetime", "incompletedatetime",
+#'   "durationdatetime", "intervaldatetime")
+#'
+#'   6) (Option only) Numeric Types - The list of classes that should be
```
Comment (bms63) [inline R/type.R:46]:
```suggestion
#'   5) Character Types (Optional)  - The list of classes that should be
#'   explicitly coerced to a XPT Character type. Default: c( "character",
#'   "char", "text", "date", "posixct", "posixt", "datetime", "time",
#'   "partialdate", "partialtime", "partialdatetime", "incompletedatetime",
#'   "durationdatetime", "intervaldatetime")
#'
#'   6) Numeric Types (Optional) - The list of classes that should be
```

File: R/length.R [hunk 10]
```diff
@@ -1,18 +1,49 @@
 #' Assign SAS Length
 #'
-#' Assigns SAS length from a variable level metadata to a given data frame.
+#' Assigns SAS length from a variable level metadata to a given data frame. If a
```
Comment (bms63) [inline R/length.R:3]:
```suggestion
#' Assigns SAS length from a metadata object to a given data frame. If a
```

variable level metadata (VLM) is something used in dataset programming and I think users might get confused.  I think all the variable level metadata references should be replaced with metadata or metadata object

File: R/label.R [hunk 11]
```diff
@@ -1,18 +1,40 @@
 #' Assign Variable Label
 #'
 #' Assigns variable label from a variable level metadata to a given data frame.
+#' This function will give an error if a label is presented that is greater than
+#' 40 characters which isn't allowed in XPT v5. If labels aren't present for the
```
Comment (bms63) [inline R/label.R:5]:
```suggestion
#' This function will give detect if a label is greater than
#' 40 characters which isn't allowed in XPT v5. If labels aren't present for the
```
The function only errors when we tell it to.  If we set as warn or message then it just prints to the console the warning or the message

File: R/label.R [hunk 12]
```diff
@@ -1,18 +1,40 @@
...
+#' variable it will be assigned an empty character value. Labels are stored in
+#' the 'label' attribute of the column.
```
Comment (bms63) [inline R/label.R:7]:
```suggestion
#' the 'label' attribute of the dataframe.
```

File: R/label.R [hunk 13]
```diff
@@ -1,18 +1,40 @@
...
+#'
+#' @inheritParams xportr_length
+#'
+#' @section Messaging: `label_log` is the primary messaging tool for
+#'   `xportr_label`. If there are any columns present in the '.df' that are not
+#'   noted in the metadata, they cannot be assigned a label and a message will
```
Comment (bms63) [inline R/label.R:13]:
```suggestion
#' @section Messaging: `label_log()` is the primary messaging tool for
#'   `xportr_label()`. If there are any columns present in the '.df' that are not
#'   noted in the metadata, they cannot be assigned a label and a message will
```

File: R/label.R [hunk 14]
```diff
@@ -1,18 +1,40 @@
...
+#'   be a metacore object, or a data.frame containing the data listed below. If
+#'   metacore is used, no changes to options are required.
+#'
+#'   For data.frame 'metadata' arguments three columns must be present:
+#'
+#'   1) Domain Name - passed as the 'xportr.domain_name' option. Default:
+#'   "dataset". This is the column subset by the 'domain' argument in the
+#'   function. 2) Variable Name - passed as the 'xportr.variable_name' option.
+#'   Default: "variable". This is used to match columns in '.df' argument and
+#'   the metadata. 3) Variable Label - passed as the 'xportr.label' option.
```
Comment (bms63) [inline R/label.R:31]: Formatting is a bit wonky here as well 
image

File: R/label.R [hunk 15]
```diff
@@ -1,18 +1,40 @@
...
-#'   Found. Options are 'stop', 'warn', 'message', and 'none'
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
 #'
 #' @return Data frame with label attributes for each variable.
-#' @family metadata functions
-#' @seealso [xportr_df_label()], [xportr_format()] and [xportr_length()]
+#'
 #' @export
 #'
 #' @examples
```
Comment (bms63) [inline R/label.R:40]: Can we get an example of greater than 40 label or missing label with verbose = warn

File: R/format.R [hunk 16]
```diff
@@ -1,18 +1,28 @@
...
+#'   For data.frame 'metadata' arguments three columns must be present:
+#'
+#'   1) Domain Name - passed as the 'xportr.domain_name' option. Default:
+#'   "dataset". This is the column subset by the 'domain' argument in the
+#'   function. 2) Format Name - passed as the 'xportr.format_name' option.
+#'   Default: "format". Character values to update the 'format.sas' attribute of
+#'   the column. This is passed to `haven::write` to note the format. 3)
+#'   Variable Name - passed as the 'xportr.variable_name' option. Default:
+#'   "variable". This is used to match columns in '.df' argument and the
+#'   metadata.
```
Comment (bms63) [inline R/format.R:24]: Itemize List is wonky

File: R/format.R [hunk 17]
```diff
@@ -1,18 +1,28 @@
...
-#'   `.df` value as a subset condition.
-#' @param verbose The action the function takes when a variable label isn't.
-#'   found. Options are 'stop', 'warn', 'message', and 'none'
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
+#' @inheritParams xportr_length
 #'
 #' @return Data frame with `SASformat` attributes for each variable.
-#' @family metadata functions
-#' @seealso [xportr_label()], [xportr_df_label()] and [xportr_length()]
+#'
```
Comment (bms63) [inline R/format.R:10]: Should we not that messaging is currently not available for this function??

File: R/order.R [hunk 18]
```diff
@@ -1,13 +1,43 @@
 #' Order variables of a dataset according to Spec
 #'
-#' @param .df A data frame of CDISC standard.
-#' @param metadata A data frame containing variable level metadata.
-#' @param domain A character value to subset the `.df`. If `NULL`(default), uses
-#'   `.df` value as a subset condition.
-#' @param verbose Option for messaging order results
-#' @param metacore `r lifecycle::badge("deprecated")` Previously used to pass metadata now renamed with `metadata`
+#' The `dplyr::arange` function is used to order the columns of the dataframe.
```
Comment (bms63) [inline R/order.R:3]:
```suggestion
#' The `dplyr::arrange()` function is used to order the columns of the dataframe.
```

File: R/order.R [hunk 19]
```diff
@@ -1,13 +1,43 @@
...
+#'
+#' @section Metadata: The argument passed in the 'metadata' argument can either
+#'   be a metacore object, or a data.frame containing the data listed below. If
+#'   metacore is used, no changes to options are required.
+#'
+#'   For data.frame 'metadata' arguments three columns must be present:
+#'
+#'   1) Domain Name - passed as the 'xportr.domain_name' option. Default:
+#'   "dataset". This is the column subset by the 'domain' argument in the
+#'   function. 2) Variable Name - passed as the 'xportr.variable_name' option.
```
Comment (bms63) [inline R/order.R:34]: wonky

**Review by bms63 — CHANGES_REQUESTED** [review]
Awesome!! This feels like a real package now with proper documentation

I like the use of `inheritParams` to minimize updating the params.

## Notes
*  I think we should have examples that showcase the messages or warnings for each function - this is nice to see in action.

* We references a lot of the options usage in the documentation.  I think we should link out to the deep-dive.Rmd for users to learn more about the options.

* We should avoid the use of value level metadata as it means something different for ADaM programming.  I find it a confusing term.  We should use just metadata or metadata object.

File: R/type.R (hunk 9 above)
Comment (elimillera) [reply R/type.R:46]: Here I'm noting that its a option rather than a metadata column.

File: R/format.R (hunk 16 above)
Comment (elimillera) [reply R/format.R]: Fixed

File: R/label.R (hunk 15 above)
Comment (elimillera) [reply R/label.R:44]: I think that would be too verbose for function documentation and would be a good example in the deep-dive vigentte

File: R/label.R (hunk 14 above)
Comment (elimillera) [reply R/label.R]: Fixed

File: R/length.R (hunk 5 above)
Comment (elimillera) [reply R/length.R:53]: I think that is too verbose for function documentation and is a good example for vignettes and the README

File: R/order.R (hunk 19 above)
Comment (elimillera) [reply R/order.R]: Fixed

File: R/format.R (hunk 17 above)
Comment (elimillera) [reply R/format.R:10]: I think the lack of 'verbose' argument and no messaging section notes that

File: R/type.R (hunk 7 above)
Comment (elimillera) [reply R/type.R]: updated

File: R/length.R (hunk 5 above)
Comment (bms63) [reply R/length.R:53]: haha!! good point

File: R/type.R (hunk 9 above)
Comment (bms63) [reply R/type.R:46]: ah that makes sense.

File: R/format.R [hunk 20]
```diff
@@ -31,7 +44,6 @@
 xportr_format <- function(.df,
                           metadata = NULL,
                           domain = NULL,
```
Comment (averissimo) [inline R/format.R:46]: Shouldn't there be a `xportr.format_verbose` parameter?

```suggestion
                          domain = NULL,
                          verbose = getOption("xportr.format_verbose", "none"),
```

File: R/df_label.R [hunk 21]
```diff
@@ -27,12 +37,15 @@
 #' )
 #'
 #' adsl <- xportr_df_label(adsl, metadata)
-xportr_df_label <- function(.df, metadata = NULL, domain = NULL, metacore = deprecated()) {
+xportr_df_label <- function(.df,
+                            metadata = NULL,
+                            domain = NULL,
```
Comment (averissimo) [inline R/df_label.R:42]: Even if not used in the function's body, it is documented in the man page

```suggestion
                            domain = NULL,
                            verbose = getOption("xportr.df_label_verbose", "none"),
```

Comment (elimillera) [reply R/df_label.R:42]: @averissimo I would agree that they should be consistent in their API, but I would say the API is driven by the functionality of the function itself. The functionality between the 'messaging' and 'non-messaging' functions are inconsistent (which should be updated in the next release) so I would say the API being different is ok.

Let me know what people think in the meeting. I'm happy to revert back if people disagree with the above.

**Pull request merged**